pkg/controller/component/dependent.go

```diff
@@ -4,14 +4,78 @@ import (
 	"context"
 	routev1 "github.com/openshift/api/route/v1"
 	"github.com/snowdrop/component-operator/pkg/apis/component/v1alpha2"
+	"github.com/snowdrop/component-operator/pkg/controller"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/api/extensions/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+type DependentResource interface {
+	Prototype() runtime.Object
+	PrototypeAsObject() v1.Object
```

Comment:
Why do we need a different object for the Openshift route ? Will this code be working when deployed on k8s where route api does not exists (= specific api of openshift) ?

Reply:
I don't understand this comment. Why are you talking about route? This code has nothing to do with routes.

Reply:
Forget my remark as I mixed api import of v1 and routev1 ;-)